src/lib/lit-extended.ts

```diff
@@ -77,12 +77,20 @@ export class PropertyPart extends AttributePart {
     if (s.length === 2 && s[0] === '' && s[1] === '') {
       // An expression that occupies the whole attribute value will leave
       // leading and trailing empty strings.
+      if (values[startIndex] === this._previousValues[startIndex]) {
+        return;
```

Comment:
In NodePart we only do this for primitives, in case a nested value changed. Since complex objects can flow to here, and could have a different ToString value even if referentially identical, we need the same check.